lib/active_model/serializer.rb

```diff
@@ -57,9 +56,9 @@ def self.digest_caller_file(caller_line)
     end
 
     def self.inherited(base)
+      inherit_attributes(base)
+      inherit_associations(base)
```

Comment:
What wasn't right in the design was that the `Associations` module was overriding the `inherited` method of the `Serializer` class, which worked fine as long as only one module was doing that. I'm all up for suggestion as to making this cleaner though.